tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Remoting/RemotingIntegration.cs

```diff
@@ -0,0 +1,100 @@
+// <copyright file="RemotingIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NETFRAMEWORK
+using System;
+using System.Runtime.Remoting.Messaging;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Remoting
+{
+    internal static class RemotingIntegration
+    {
+        internal const IntegrationId IntegrationId = Configuration.IntegrationId.Remoting;
+        internal const string IntegrationName = nameof(Configuration.IntegrationId.Remoting);
+
+        internal const string Major4 = "4";
+
+        private const string ClientOperationName = "dotnet_remoting.client.request";
+        private const string ServerOperationName = "dotnet_remoting.server.request";
+        private const string ServiceName = "remoting";
+
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(RemotingIntegration));
+
+        internal static Scope CreateServerScope(IMessage msg, SpanContext spanContext)
+        {
+            var tracer = Tracer.Instance;
+
+            if (!tracer.Settings.IsIntegrationEnabled(IntegrationId))
+            {
+                // integration disabled, don't create a scope, skip this trace
+                return null;
+            }
+
+            Scope scope = null;
+
+            try
+            {
+                var tags = new RemotingServerTags();
+                scope = tracer.StartActiveInternal(ServerOperationName, parent: spanContext, tags: tags);
+                var span = scope.Span;
+
+                var methodMessage = msg as IMethodMessage;
+                tags.MethodName = methodMessage?.MethodName;
+                // tags.MethodService = methodMessage?.MethodMes
```

Comment:
Is something missing here?

Reply:
Whoops, I was trying to get an RPC Service name from this, but I was unable to find anything. If we find a reliable way to add the `rpc.service` tag, we can do so later.